### PR TITLE
Answer 1 projection

### PR DESCRIPTION
**Pull request by SephShom**
## Checklist for challenge submission

- [ ] Start your PR message with Answer:${challenge_number}

### CODE REVIEW

File: apps/angular/projection/src/app/ui/card/card.component.ts [hunk 1]
```diff
@@ -1,61 +1,55 @@
...
-    if (this.type === CardType.TEACHER) {
-      this.teacherStore.addOne(randTeacher());
-    } else if (this.type === CardType.STUDENT) {
-      this.studentStore.addOne(randStudent());
-    }
-  }
+export class CardComponent<T> {
+  @Input() list: T[] | null = null;
+  @Output() actionClicked = new EventEmitter();
+  @ContentChild(CardListItemDirective) rowTemplate!: CardListItemDirective<T>;
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/ui/card/card.component.ts:54]: awesome work

File: apps/angular/projection/src/app/component/student-card/student-card.component.ts [hunk 2]
```diff
@@ -1,40 +1,55 @@
...
-  students: Student[] = [];
-  cardType = CardType.STUDENT;
+  private http = inject(FakeHttpService);
+  private store = inject(StudentStore);
 
-  constructor(
-    private http: FakeHttpService,
-    private store: StudentStore,
-  ) {}
+  public students$ = this.store.students$;
```
Comment (tomalaforge) [inline apps/angular/projection/src/app/component/student-card/student-card.component.ts:42]: To go a bit further, you should transform this to signal to benefit and get use to the future of angular

**Comment by tomalaforge** [comment]
great work. 👍

File: apps/angular/projection/src/app/ui/card/card.component.ts (hunk 1 above)
Comment (SephShom) [reply apps/angular/projection/src/app/ui/card/card.component.ts:54]: Thank you for taking the time to review it

**Comment by tomalaforge** [comment]
I don't know what you did, but you messed up. You have 219 changes files.

**Comment by SephShom** [comment]
I know, broken it, somehow sync functionality enforce me to accept a merge and everything blow up. Maybe I should restart all the process

**Pull request closed without merging**